File: src/main/java/com/projectone/services/PostService.java
package com.projectone.services;

import com.projectone.dto.Post;
import com.projectone.entities.PostEntity;

import java.security.Principal;
import java.util.List;

public interface PostService {
  List<Post> getAllPostByUser(Long userId);
  Post save(Principal principal, PostEntity postEntity);
  Post getPostByUserId(Long userId, Long postId);
  Post getPostById(Long postId);
  Post updatePost(Long postId, PostEntity postEntity);
  void deletePostById(Long postId);
}

File: src/front/src/serveces/PostService.js
import http from "axios";

class PostService {
    getAllByUserId(userId) {
        return http.get(`api/posts?userId=${userId}`);
    }

    get(postId) {
        return http.get(`api/posts/${postId}`)
    }

    createPost(data, userId) {
        return http.post(`api/posts?userId=${userId}`, data)
    }
}

export default new PostService();